### DISCUSSION TITLE
Implementing a login page

### DISCUSSION THREAD

**Discussion by Edu-Vidal** [hoppscotch/hoppscotch#2909] [Q&A] [answered]
Hello my fellow developers!
I'm currently implementing a hoppscotch version for internal usage. And some legal issues arose around the possibility of anonymous users abusing requests under our domain. Currently considering implementing a logged in requirement for making requests.
Are there any straight forward ways to implement this? Have anyone dealt with this already?

Not familiar with Vue, but greatful for any help given 😅.

**Comment by Edu-Vidal** [comment]
Ended up copying most of the "join-team" page design to the default layout. Not the prettiest solution, that's for sure, but good enough.

**Comment by sjugraj** [comment]
Could you please guide me on how to enforce a login requirement before someone can use the tool? Thank you.

**Comment by dancabper** [comment]
Hi Edu, I think you can always use Google's external reCaptcha service. This new functionality that you have created is very good since it helps companies that have their internal services not have the need to open ports externally to be able to communicate through callbacks to github, microsoft or google. Waiting for news of the completion I stay.

**Comment by liyasthomas** [comment, marked as answer]
As far as the Hoppscotch Self-Hosted instances go, we've two editions.

1. Community Edition
2. Enterprise Edition

Both self-hosted editions have an admin panel with features for users and team management. I'd advise you to check out Hoppscotch [Self-Hosted Enterprise Edition](https://hoppscotch.com/pricing) which has the ability to create Private Self Hosted Hoppscotch instances. Only admin-approved users can sign up and log in to Hoppscotch App and thereby create teams.

Imagine the use case for organizations who want to self-host Hoppscotch on their own infrastructure but want to limit their instance to their employees — Hoppscotch Self-Host Enterprise will be a good fit.

Self-hosting documentation: https://docs.hoppscotch.io/documentation/self-host/getting-started
